sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java

```diff
@@ -1696,6 +1737,17 @@ WithSystemAssignedIdentityBasedAccessOrCreate withSystemAssignedIdentityBasedAcc
             WithUnmanagedCreate withOSDiskVhdLocation(String containerName, String vhdName);
         }
 
+        /** The stage of the definition allowing to specify delete options for the network interface. */
+        interface WithNetworkInterfaceDeleteOptions {
+            /**
+             * Sets delete options for primary network interfaces.
+             *
+             * @param deleteOptions the delete options for primary network interfaces
+             * @return the next stage of the definition
+             */
+            WithCreate withPrimaryNetworkInterfaceDeleteOptions(DeleteOptions deleteOptions);
+        }
```

Comment:
This is the API that I am not certain whether it is best handled here, or should I add 2 overload at
```
            WithPrivateIP withNewPrimaryNetwork(Creatable<Network> creatable);

            WithPrivateIP withNewPrimaryNetwork(String addressSpace);
```
with 2nd parameter of `DeleteOptions`.